site/en/blog/pop-ups-theyre-making-a-resurgence/index.md

```diff
@@ -0,0 +1,729 @@
+
+![GIF of DevTools top layer support being demonstrated](https://wd.imgix.net/image/1D9D0Ls1ATa2ZPA9x2ZWrGFyZzT2/36Yck7O77zDipSNGNNbB.gif?auto=format&w=1600
+)
+
+- You get "Light Dismiss" out of the box. By that, we mean you can close the pop-up with a close signal, such as clicking outside the pop-up, keyboard-navigating to another element, or pressing the "Esc" key. Open it up again and try it out!
+
+What else do you get with pop-up? Let's take our example further. Consider this demo where we have some content on the page.
+
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'GRxLZYe',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+That floating action button has fixed positioning with a high `z-index`.
+
+```css
+.fab {
+  position: fixed;
+  z-index: 99999;
+}
+```
+
+{% Aside %}
+The floating action button could also be a pop-up.
+{% endAside %}
+
+Our pop-up content is nested in the DOM, but when you open the pop-up, it gets promoted above that fixed position element. We didn’t need to set any styles.
+
+You may also notice that our pop-up now has a `::backdrop` pseudo-element. All elements that are in the top layer get a styleable `::backdrop` pseudo-element. For our example, we've set a reduced alpha background color and a backdrop filter, which  blurs out the underlying content.
+
+{% Aside %}
+By default, the `::backdrop` on a pop-up has `pointer-events: none` set.
+{% endAside %}
+
+
+## Styling a pop-up
+
+Let's turn our attention to styling the pop-up. By default, a pop-up has a fixed position and some applied padding. It also has `display: none`. You could override this to show a pop-up. But, that wouldn't promote it to the top layer.
+
+```css
+[popup] { display: block; } 
+```
+
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'ZExZOpX',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+If you want a pop-up to be shown on load, you can use the `defaultopen` attribute.
+
+```html
+<div popup defaultopen id=”pop”>Pop-up content!</div>
+```
+
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'KKoYMmm',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+Regardless of how you promote your pop-up, once you promote a pop-up to the top layer, you may need to lay it out or position it. You can't target the top layer and do something like
+
+```css
+:open {
+  display: grid;
+  place-items: center;
+}
+```
+
+By default, a pop-up will lay out in the center of the viewport using `margin: auto`. But, in some cases, you may want to be explicit about positioning. For example:
+
+```css
+[popup] {
+  top: 50%;
+  left: 50%;
+  translate: -50%;
+}
+```
+
+If you want to lay out content inside your pop-up using CSS Grid or Flexbox, it might be wise to wrap this in an element. Otherwise, you'll need to declare a separate rule that changes the `display` once the pop-up is in the top layer. Setting it by default would have it shown by default overriding `display: none`.
```

Comment:
```suggestion
If you want to lay out content inside your pop-up using CSS grid or flexbox, it might be wise to wrap this in an element. Otherwise, you'll need to declare a separate rule that changes the `display` once the pop-up is in the top layer. Setting it by default would have it shown by default overriding `display: none`.
```